src/System.Drawing.Primitives/tests/SizeTests.netcoreapp.cs

```diff
@@ -0,0 +1,132 @@
+        }
+
+
+        [Theory]
+        [InlineData(1000, 2400.933f)]
+        [InlineData(int.MaxValue, 0.0f)]
+        [InlineData(int.MaxValue, 1.0f)]
+        [InlineData(int.MaxValue, -1.0f)]
+        [InlineData(int.MaxValue, float.MaxValue)]
+        [InlineData(int.MinValue, 0.0f)]
+        [InlineData(int.MinValue, 1.0f)]
+        [InlineData(int.MinValue, -1.0f)]
+        [InlineData(int.MinValue, float.MinValue)]
+        [InlineData(int.MaxValue, float.MinValue)]
+        [InlineData(int.MinValue, float.MaxValue)]
+        public void MultiplicationTestSizeFloat(int value1, float value2)
+        {
+            Size sz1 = new Size(value1, value1);
+            SizeF mulExpected;
+
+            unchecked
+            {
+                mulExpected = new SizeF(value1 * value2, value1 * value2);
+            }
+
+            Assert.Equal(mulExpected, sz1 * value2);
+            Assert.Equal(mulExpected, value2 * sz1);
+        }
+
+        [Fact]
+        public void DivideByZeroChecks()
+        {
+            Size size = new Size(100, 100);
+            Assert.Throws<DivideByZeroException>(() => size / 0);
+
+            float invDiv = 1 / 0.0f;
+            SizeF expectedSizeF = new SizeF(float.PositiveInfinity, float.PositiveInfinity);
+            Assert.Equal(expectedSizeF, size * invDiv);
+        }
+
+        [Theory]
+        [InlineData(0, 1)]
+        [InlineData(1, 1)]
+        [InlineData(-1, 1)]
+        [InlineData(1, -1)]
+        [InlineData(-1, -1)]
+        [InlineData(int.MaxValue, int.MaxValue)]
+        [InlineData(int.MaxValue, int.MinValue)]
+        [InlineData(int.MinValue, int.MaxValue)]
+        [InlineData(int.MinValue, int.MinValue)]
+        [InlineData(int.MaxValue, 1)]
+        [InlineData(int.MinValue, 1)]
+        [InlineData(int.MaxValue, -1)]
+        public void DivideTestSizeInt(int value1, int value2)
+        {
+            Size size = new Size(value1, value1);
+            Size expected;
+
+            int invDiv = 1 / value2;
+            expected = new Size(value1 * invDiv, value1 * invDiv);
+
+            Assert.Equal(expected, size / value2);
+        }
+
+        [Theory]
+        [InlineData(0, 1.0f)]
+        [InlineData(1, 1.0f)]
+        [InlineData(-1, 1.0f)]
+        [InlineData(1, -1.0f)]
+        [InlineData(-1, -1.0f)]
+        [InlineData(int.MaxValue, float.MaxValue)]
+        [InlineData(int.MaxValue, float.MinValue)]
+        [InlineData(int.MinValue, float.MaxValue)]
+        [InlineData(int.MinValue, float.MinValue)]
+        [InlineData(int.MaxValue, 1.0f)]
+        [InlineData(int.MinValue, 1.0f)]
+        [InlineData(int.MaxValue, -1.0f)]
+        [InlineData(int.MinValue, -1.0f)]
+        public void DivideTestSizeFloat(int value1, float value2)
+        {
+            SizeF size = new SizeF(value1, value1);
+            SizeF expected;
+
+            float invDiv = 1.0f / value2;
+
+            expected = new SizeF(value1 * invDiv, value1 * invDiv);
```

Comment:
@tannergooding hmmm, I think I still tend to disagree. Because the focus of the test is to validate behavior or operator / and not that of floating point division. For example the ArithmeticTest in `Size`. The `Size.Add` and `operator +` is tested in same whole - part concept way and not in terms of other arithmetic operations such as `a + b` being tested as `a - (-1 * b)`...

Coming back to `operator /`, whatever mess might happen by performing optimization or refactoring would happen identically inside and outside and hence behavior of operator / would still be consistent. But, probably, there would be another test case, say, on floating point type that would validate result of division being within acceptable limits vis-a-vis reciprocal multiplication... which would flag off an error should there be any discrepancy due to any issues...